src/coreclr/tests/src/Directory.Build.targets

```diff
@@ -199,6 +199,12 @@
   <Target Name="UpdateReferenceItems"
           BeforeTargets="BeforeResolveReferences"
   >
+   <MSBuild Projects="$(MSBuildProjectFullPath)"
```

Comment:
Why can't we use normal target dependencies here?

Reply:
For tests that reference System.Private.CoreLib directly and don't reference the libraries, the test build for that specific test doesn't need to depend on the outputs of the libraries build. By implementing this via a conditional `<MSBuild>` task instead of a conditional `<ItemGroup>`, we don't need a libraries build for something referencing S.P.CL directly. This isn't strictly required now, but it made more sense in an earlier iteration of this PR where the `ResolveLibrariesFromLocalBuild` target could rebuild the libraries if they were out of date.

Reply:
Hmm, does that mean that those tests never reference other framework libraries, e.g. System.IO or such?

Reply:
Yep. They reference only `System.Private.CoreLib`. There's a small number of these tests, mainly in the Interop and AppModel space for APIs that are not exposed in any public contracts but are called directly from the hosting layer.

It also used to be used to test APIs in CoreCLR that we hadn't yet exposed via CoreFX so we could add tests to a S.P.CL API in the same PR as adding the API (once again mainly in the Interop/App Model space since many of our tests include native components or need to be in their own executable and as a result lived in the CoreCLR test tree). With live-live being enabled, this usage is no longer relevant.

Reply:
I see, thanks for clarification. In fact I hit trouble with these tests multiple times during the switch-over to the runtime repo and in the live-live pipeline, the special conditions in the scripts for dealing with them are just so annoying. If there's any chance to make this somewhat simpler now we're in the combined repo, I would be super-happy.

Reply:
I think we've already reduced the set of tests using this functionality to as few as possible, but we can add that as another thing to check over once we get live-live working.

Reply:
I think we can safely ignore this for now as these tests are certain to jump up during any future script refactoring so they act as an automated work item.